### ISSUE TITLE
Cannot set size encoding using an expr

### ISSUE THREAD

**Issue by aantn** [vega/vega-lite#7720]
Hi, I'm trying to set the size encoding using a param. I've excluded the param in the code below as I'm having trouble one step earlier when I try to specify the size encoding using an expr.

I'm not sure if this is a bug or if I'm using datum/expr here wrong. I've tried to read all relevant docs.

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "description": "A Wilkinson Dot Plot",
  "height": 100,
  "data": {
    "values": [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 3, 3, 4, 4, 4, 4, 4, 4]
  },
  "transform": [{"window": [{"op": "rank", "as": "id"}], "groupby": ["data"]}],
  "mark": {"type": "circle", "opacity": 1},
  "encoding": {
    "x": {"field": "data", "type": "ordinal"},
    "y": {"field": "id", "type": "ordinal", "axis": null, "sort": "descending"},
    "size": {"datum": {"expr": "200"}}
  }
}
```

[Example in Vega-Editor](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEzjQATjRyZ289AEEABAHUaggNY1ZM2YYAibTIYAKgx4oxwaZLGgCMAAwBSspImCiooMRIggxqaADafgrJqSnpaWkATAo5OQDMCoUALAql5WVlALoAvkqY6kjWAGZs6giJoADuVspsXZ0gbDhoII2y5h5IEKM0yiA1VUpk6mwMOEwAnokqYSi1SyDI6pORIJibOHCjUDTqUILXSsNItxf+dSBwslBsylZkNCgAAeQJAzRocEE83QoXCHguV1GbX+shiCyU2zOEKhMJAcwRl2u6BRVnRSiQwJoM1QsgYgkEShk6kwo1U0G+qMBnwgNAAXsTQHCGB0znBgTh1KMskEFjV5UA).

**Comment by domoritz** [comment]
Thanks for the report. Can you provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) that demonstrates the issue?

**Comment by aantn** [comment]
Yes, here you go:

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "data": {
    "values": [1]
  },
  "mark": {"type": "circle"},
  "encoding": {
    "x": {"field": "data", "type": "ordinal"},
    "size": {"datum": {"expr": "200"}}
  }
}
```

[Vega Editor Link](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUMSSCGcCGgDaARgC6AXyXIATgGs0oTAE8ccNCCg1rUQW8sg4WSg2ZRpZMgcQAA9IgDMaOEFld1V1RRBnV3c2a1DZHRB-CBoALzdNFTUGBEi4KJxrdwAmAAYWwvNOoA)

**Comment by domoritz** [comment]
Thank you for the example. We don't translate the `expr` to a signal correctly. This should be fixed by e.g. #7463.

However, just to make sure, do you want to use `datum` or `value`? The latter works. [Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUMSSCGcCGgDaARgC6AXyXIATgGs0oTAE8ccNCCg1rUQW8sg4WSg2ZRpZMgcQAA9IgDMaOEFld1V1RRBnV3c2a1DZHRB-CBoALzdNEG1dctA4KJxrdwAmAAYWwvNOoA)

**Comment by aantn** [comment]
Yes, that works. Thank you.

I would also recommend adding an example to https://vega.github.io/vega-lite/docs/parameter.html showing how you can use a parameter with `value: {expr: }` inside an encoding. That is what I'm actually tring to do here and there were no examples in the docs so I assumed you should use datum for that.

**Comment by domoritz** [comment]
The distinction between datum and value should should be in a different place imo. We will reconsider how we document expr when it works everywhere consistently.

**Comment by aantn** [comment]
Absolutely, but it would be nice to have at least one example on how to use non-selection parameters inside `encoding`. All the other examples on that page involve using non-selection parameters in `mark` or using selection parameters in `encoding`. There is no example for using non-selection parameters in `encoding`.
